packages/tools/src/tools/segmentation/RectangleROIStartEndThresholdTool.ts

```diff
@@ -561,29 +604,29 @@ class RectangleROIStartEndThresholdTool extends RectangleROITool {
       numSlicesToPropagate * spacingInNormal
     );
 
-    const halfSpacingInNormalDirection = spacingInNormal / 2;
-    // Loop through imageIds of the imageVolume and find the one that is closest to endPos
-    const { imageIds } = imageVolume;
-    let imageIdIndex;
-    for (let i = 0; i < imageIds.length; i++) {
-      const imageId = imageIds[i];
+    const imageIdIndex = this._getImageIdIndex(imageVolume,endPos,viewPlaneNormal);
 
-      const { imagePositionPatient } = metaData.get(
-        'imagePlaneModule',
-        imageId
-      );
-
-      const dir = vec3.create();
-      vec3.sub(dir, endPos, imagePositionPatient);
-
-      const dot = vec3.dot(dir, viewPlaneNormal);
+    return imageIdIndex;
+  }
 
-      if (Math.abs(dot) < halfSpacingInNormalDirection) {
-        imageIdIndex = i;
-      }
+  _getImageIdIndex(
+    imageVolume: Types.IImageVolume,
+    pos: vec3,
+    viewPlaneNormal: Types.Point3,
+  ): number | undefined {
+    const { imageData } = imageVolume;
+    const imageIdIndex = imageData.worldToIndex([pos[0],pos[1],pos[2]])
+
+    const mprValues = CONSTANTS.MPR_CAMERA_VALUES
+    if (this._arraysEqual(viewPlaneNormal, mprValues.axial.viewPlaneNormal)) {
+      return Math.round(imageIdIndex[2]);
+    } else if (this._arraysEqual(viewPlaneNormal, mprValues.sagittal.viewPlaneNormal)) {
+      return Math.round(imageIdIndex[0]);
+    } else if (this._arraysEqual(viewPlaneNormal, mprValues.coronal.viewPlaneNormal)) {
+      return Math.round(imageIdIndex[1]);
+    } else {
+      return undefined;
     }
```

Comment:
I'm not sure if this is a correct logic, basically the volume has x,y,z , but it might be a sagittal acquistion volume, but still the z is the depth though sagittal. The logic is more complext than this, example of such volume can be found in this study https://viewer-dev.ohif.org/viewer?StudyInstanceUIDs=1.3.6.1.4.1.25403.345050719074.3824.20170125095438.5

Reply:
Yes I agree this may not be the good solution but I didn't find how to get the slice we project to get without checking in what plane we are. Maybe you can help us with that.